Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
+}
+
+bool Gun::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target,id_ptr_on<Human> shooter)
+{	
+    if(Handle_Ammo())
+    {
+      if(GetOwner()->GetOwner().valid())
+      {
+          id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+          p->MakeMovementPattern(target,shooter);
+      }
+             //       Create<Item>(Bullet_Casing::T_ITEM_S(),GetOwner);	
+    }
+    else
+    {
+          PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
```

Comment:
Well, this one will not work for sure. If my variant is not working then it is a bug.
The chain of ownership:
`Gun` -> `Human` -> `CubeTile`. Also it is possible that it will be
`Gun` -> `Human` -> `Locker` -> `CubeTile` (when human is inside a locker)